pkg/analysis/passes/safelinks/safelinks_test.go

```diff
@@ -0,0 +1,246 @@
+package safelinks
+
+import (
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/metadata"
+	"github.com/grafana/plugin-validator/pkg/testpassinterceptor"
+)
+
+const pluginId = "test-plugin-panel"
+
+func TestNoAPIKey(t *testing.T) {
+	// Temporarily unset the API key
+	originalKey := os.Getenv("WEBRISK_API_KEY")
+	os.Unsetenv("WEBRISK_API_KEY")
+	defer func() {
+		if originalKey != "" {
+			os.Setenv("WEBRISK_API_KEY", originalKey)
+		}
+	}()
+
+	var interceptor testpassinterceptor.TestPassInterceptor
+	pass := &analysis.Pass{
+		RootDir: filepath.Join("./"),
+		ResultOf: map[*analysis.Analyzer]interface{}{
+			metadata.Analyzer: []byte(
+				`{"ID": "` + pluginId + `", "info": {"links": [{"name": "Test Link", "url": "https://example.com"}]}}`,
+			),
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	_, err := Analyzer.Run(pass)
+	require.NoError(t, err)
+	require.Empty(t, interceptor.Diagnostics)
+}
+
+func TestSafeLink(t *testing.T) {
+	webriskApiKey := os.Getenv("WEBRISK_API_KEY")
+	require.NotEmpty(t, webriskApiKey, "API key should not be empty")
```

Comment:
this will make the test fail in build time for environments that don't define it. (e.g. some local environments)

What we've done with other tests is to skip the test if the env is not defined.